pkg/propertychecker/azure/checker.go

```diff
@@ -0,0 +1,167 @@
+//
+//Copyright (c) Microsoft Corporation.
+//Licensed under the MIT license.
+
+// Package azure provides property checkers for Azure-specific cluster requirements.
+// It checks whether the cluster can meet the requirement defined by the property selector.
+package azure
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	computev1 "go.goms.io/fleet/apis/protos/azure/compute/v1"
+	"go.goms.io/fleet/pkg/clients/azure/compute"
+	"go.goms.io/fleet/pkg/utils/labels"
+)
+
+const (
+	// maxVMInstanceCapacity defines the maximum allowed VM instance capacity for SKU capacity requirements.
+	// This limit is set to prevent excessive resource requests and potential quota issues.
+	// The value is constrained to a reasonable upper bound of 200 for most production workloads.
+	// Upper bound is enforced after adjusting for operator semantics.
+	maxVMInstanceCapacity = 200
+)
+
+// PropertyChecker provides Azure-specific property validation for member clusters.
+// It validates compute requirements to determine if clusters
+// can meet the specified property selector requirements.
+type PropertyChecker struct {
+	// vmSizeRecommenderClient is the Azure compute client used to generate VM size recommendations
+	// and validate SKU capacity requirements.
+	vmSizeRecommenderClient compute.AttributeBasedVMSizeRecommenderClient
+}
+
+// NewPropertyChecker creates a new PropertyChecker with the given client.
+// The vmSizeRecommenderClient is used to validate SKU capacity requirements.
+func NewPropertyChecker(vmSizeRecommenderClient compute.AttributeBasedVMSizeRecommenderClient) *PropertyChecker {
+	return &PropertyChecker{
+		vmSizeRecommenderClient: vmSizeRecommenderClient,
+	}
+}
+
+// CheckIfMeetSKUCapacityRequirement validates whether a member cluster can meet the specified
+// SKU capacity requirement. It extracts the required SKU and capacity from the property selector
+// requirement and checks to determine if the cluster's Azure subscription
+// and location can provision the requested VM instances.
+//
+// The cluster must have both Azure location and subscription ID labels configured.
+// Returns true if the SKU capacity requirement can be met, false otherwise.
+func (s *PropertyChecker) CheckIfMeetSKUCapacityRequirement(
+	cluster *clusterv1beta1.MemberCluster,
+	req placementv1beta1.PropertySelectorRequirement,
+	sku string,
```

Comment:
where is this sku coming from?  IIRC. req includes both the sku and the capacity (sku in the name while capacity in the value). I wonder why we process the sku info outside of this function while parse the capacity inside this function?

Reply:
https://github.com/Azure/fleet/pull/1210/commits/52ad7d78c09467aaf5fb3f697239d35fbdcca40e#diff-909e9d3ff38befb3e868379f89427c9f606366ca40d4f997f4bf0f0668a68b8a

we need to first figure out whether it's azure SKUproperty first before calling this.